pkg/controller/controller_utils.go

```diff
@@ -942,16 +942,6 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
 	}
 
-	newTaints := newNode.Spec.Taints
```

Comment:
Hmmm i now see, but the thing is that the newNode comes only from RemoveTaint and AddOrUpdateTain (check done before submtting PR, as said here #46330  ).
We deepCopy there and we modify only the taints, that's why i said that we do not need to deepCopy, that's why i submitted this PR :)